compiler/modules/front/resolveTypes.bal

```diff
@@ -28,6 +28,41 @@ function resolveTypes(ModuleSymbols mod) returns ResolveTypeError? {
     foreach var defn in mod.defns {
         check resolveDefn(mod, defn);
     }
+    if !mod.tc.env.isReady() {
+        // This should never happen
+        panic err:impossible("type environment is not ready");
+    }
+    foreach var { semType, modDefn, td } in mod.deferredEmptinessChecks {
+        // XXX When we can give multiple errors we should check all the deferred emptiness checks
+        check testTypeForFiniteShape(mod, semType, modDefn, td);
+    }
+    if mod.emptySourceTypeIndices.length() != 0 {
```

Comment:
This doesn't need to be in `mod`. We need only a single location not the list of indices.

Reply:
fixed